app/javascript/src/components/Dashboard/Notes/NewNoteForm.jsx

```diff
@@ -21,18 +63,63 @@ export default function NewNoteForm({ onClose, refetch }) {
     <Formik
       initialValues={{
         title: "",
-        description: ""
+        tags: null,
+        description: "",
+        contact: null,
+        addDueDateToNote: selectedNote.dueDate,
+        dueDate: null,
+        ...editableNote
       }}
       onSubmit={handleSubmit}
-      validationSchema={yup.object({
-        title: yup.string().required("Title is required"),
-        description: yup.string().required("Description is required")
-      })}
+      validationSchema={formValidationSchemas.newNoteForm}
     >
-      {({ isSubmitting }) => (
+      {({ isSubmitting, values }) => (
         <Form>
           <Input label="Title" name="title" className="mb-6" />
-          <Textarea label="Description" name="description" rows={8} />
+          <Select
+            label="Tags"
+            value={values.tags}
+            placeholder="Select an Option"
+            isDisabled={false}
+            isClearable={true}
+            isSearchable={true}
+            name="tags"
+            options={TAG_OPTIONS}
+            className="mb-6"
+          />
+          <Textarea
+            label="Description"
+            name="description"
+            rows={8}
+            className="mb-6"
+          />
+          <Select
+            label="Assigned Contact"
+            placeholder="Select an Option"
+            isDisabled={false}
+            isClearable={true}
+            isSearchable={true}
+            name="contact"
+            options={CONTACT_OPTIONS}
+            className="mb-6"
+          />
+          <div className="flex mb-6 justify-between">
+            <label htmlFor="addDueDateToNote">Add Due Date to Note</label>
+            <Switch name="addDueDateToNote" />
+          </div>
+          {values.addDueDateToNote && (
+            <Field name="dueDate">
+              {({ field: { name, value }, form }) => (
+                <DateInput
+                  minDate={selectedNote.id ? undefined : new Date()}
+                  label="Due Date"
+                  format="DD/MM/YYYY"
+                  value={value}
+                  onChange={newDate => form.setFieldValue(name, newDate)}
+                />
+              )}
+            </Field>
+          )}
```

Comment:
Rather than applying `mb-6` on each individual element, you may use `space-y-6` on the parent wrapper. 

Please refer [space between](https://tailwindcss.com/docs/space) in tailwindCSS.